features/steps/core.py

```diff
@@ -67,9 +67,11 @@ def set_config(context, config_file):
             cf.write("version: {}".format(__version__))
 
 
-@when('we open the editor and exit')
-def open_editor_and_exit_without_entering_text(context):
-    with patch('jrnl.util.get_text_from_editor', return_value=""):
+@when('we open the editor and enter')
+@when('we open the editor and enter {text}')
+def open_editor_and_enter(context, text=""):
+    text = (text or context.text)
+    with patch('subprocess.call', return_value=text):
```

Comment:
If you look at the get_text_from_editor function, it doesn't use the return value of subprocess.call, but instead lets it write something to a tempfile and then reads that tempfile. So your patched function would have to write `text` to that tmpfile. 
You can do this by creating a helper function that takes a `List[str]` (like subprocess.call), takes the last string of that list, which will be the path to the tmpfile, and then writes `text` to that file. You can then use this function with `patch('subprocess.call', side_effect=_mock_editor_function)`.

Reply:
Ah, yep, you're right. Missed that detail.

Reply:
Do you know a good way to pass `text` to that patched function? It won't be in the `subprocess.call` args, but it needs to be accessible from within the patched function that's overriding `subprocess.call`.

Reply:
You could define the function within `open_editor_and_enter()`, then it has access to `text`.

Reply:
I've made those changes in my last commit, however, the patched function doesn't ever seem to be called. Still looking into why.

Reply:
It's possible that I'm not patching the right reference to `subprocess.call`, but I've tried it with `jrnl.utils.subprocess.call` as well, but no luck.

Reply:
In travis, the function does seem to be called. The print statements show up in the test output.

Reply:
It's not happening locally for me. I've got no idea how that's possible. I guess I'll just work off the Travis runs for testing :/

Reply:
<img width="741" alt="image" src="https://user-images.githubusercontent.com/20600565/68993772-fc976000-087b-11ea-94db-b37fb3ca847f.png">

So, the empty file does get written, but for some reason, this code block in `jrnl/cli.py` is never hit:

```python
raw = util.get_text_from_editor(config, template)
if not raw:
	sys.exit()
```

Reply:
That's weird. utils doesn't do a `from subprocess import call` but instead looks up `subprocess.call` during the function call. So patching `subprocess.call` should be correct. Not sure why it's not working on your end. 

I have no idea why raw is not empty either at this point..

Reply:
Yeah, I just did that. Waiting for Travis.